daemon/api.go

```diff
@@ -316,7 +319,20 @@ func getSnapInfo(c *Command, r *http.Request) Response {
 		channel = localSnap.Channel
 	}
 
-	remoteSnap, _ := remoteRepo.Snap(name, channel, nil)
+	var authenticator store.Authenticator
+	overlord := c.d.overlord
+	state := overlord.State()
+	state.Lock()
+	user, err := UserFromRequest(state, r)
+	state.Unlock()
+	if err != nil {
+		return InternalError("%v", err)
+	}
+	if user != nil {
```

Comment:
If UserFromRequest returns err == nil, the user value should always be nil. We asked "give me a user" and it said "no problem". It's lying if the user is nil.